src/main/java/com/Novi/TechItEasy/services/CI_ModuleService.java

```
@@ -0,0 +1,116 @@
+
+        if (fetchedCiModule.isPresent()) {
+            CI_ModuleDto dto = new CI_ModuleDto();
+            DtoConverters.CI_ModuleDtoConverter(fetchedCiModule.get(), dto);
+            return dto;
+
+        } else {
+            throw new RecordNotFoundException("We hebben geen ci module met dit ID.");
+        }
+    }
+
+
+    /// For creating one new ci module in database /////
+    public CI_ModuleDto createCiModule(CI_ModuleInputDto ci_module) {
+
+        CI_Module newCiModule = new CI_Module();
+        DtoConverters.CI_ModuleInputDtoConverter(newCiModule, ci_module);
+
+        ci_moduleRepository.save(newCiModule);
+
+        CI_ModuleDto dto = new CI_ModuleDto();
+        DtoConverters.CI_ModuleDtoConverter(newCiModule, dto);
+
+
+        return dto;
+    }
+
+
+    ///// For changing ci module in database /////
+    public CI_ModuleDto changeCiModule(Long id, CI_ModuleDto ci_moduleDto) {
+
+        Optional<CI_Module> databaseCiModule = ci_moduleRepository.findById(id);
+
+        if (databaseCiModule.isPresent()) {
+            CI_Module fetchedCiModule = databaseCiModule.get();
+
+            if (ci_moduleDto.name != null) {
+                fetchedCiModule.setName(ci_moduleDto.name);
+            }
+            if (ci_moduleDto.type != null) {
+                fetchedCiModule.setType(ci_moduleDto.type);
+            }
+            if (ci_moduleDto.price != null) {
+                fetchedCiModule.setPrice(ci_moduleDto.price);
+            }
+
+            ci_moduleRepository.save(fetchedCiModule);
+            DtoConverters.CI_ModuleDtoConverter(ci_moduleRepository.findById(id).get(), ci_moduleDto);
+
+            return ci_moduleDto;
+
+        } else {
+            throw new RecordNotFoundException("We hebben geen ci module met dit ID.");
+        }
+    }
+
+
+    ///// For deleting ci module from database /////
+    public String deleteCiModule(Long id) {
+        if (ci_moduleRepository.existsById(id)) {
```

Heel goed dat je hier ook voor id checkt. Je krijgt geen foutmelding als je een verkeerd id verwijderd (fails silently), maar op deze manier kun je wel naar de client communiceren dat deze een verkeerd id heeft ingevoerd.